crates/core/src/protocols/fri/common.rs

```diff
@@ -139,36 +141,34 @@ where
 	P: PackedField<Scalar = F>,
 {
 	// Preconditions
-	debug_assert!(fold_challenges.len() <= ntt.log_domain_size() + log_batch_size);
-	debug_assert_eq!(len_packed_slice(values), 1 << (log_batch_size + fold_challenges.len()));
-	debug_assert_eq!(tensor.len(), 1 << log_batch_size);
-	debug_assert!(scratch_buffer.len() >= 2 * (values.len() >> log_batch_size));
-
-	// TODO: handle the case when log_batch_size is not a multiple of P::LOG_WIDTH
-	assert!(log_batch_size >= P::LOG_WIDTH);
-
-	// There are two types of mixing we do in this loop. Buffer 1 is populated with the
-	// folding of symbols from the interleaved codewords into a single codeword. These
-	// values are mixed as a regular tensor product combination. Buffer 2 is then
-	// populated with `fold_chunk`, which folds a coset of a codeword using the FRI
-	// folding algorithm.
-	let (buffer1, buffer2) = scratch_buffer.split_at_mut(1 << fold_challenges.len());
-
-	for (interleave_chunk, val) in values
-		.chunks(1 << (log_batch_size - P::LOG_WIDTH))
-		.zip(buffer1.iter_mut())
-	{
-		*val = inner_product_unchecked(
-			PackedField::iter_slice(interleave_chunk),
-			tensor.iter().copied(),
-		);
-	}
-
-	if fold_challenges.is_empty() {
-		buffer1[0]
+	debug_assert!(fold_challenges.len() <= log_len);
+	debug_assert!(log_len <= ntt.log_domain_size());
+	debug_assert_eq!(
+		values.len(),
+		1 << (fold_challenges.len() + log_batch_size).saturating_sub(P::LOG_WIDTH)
+	);
+	debug_assert_eq!(tensor.len(), 1 << log_batch_size.saturating_sub(P::LOG_WIDTH));
+	debug_assert!(scratch_buffer.len() >= 1 << fold_challenges.len());
+
+	let scratch_buffer = &mut scratch_buffer[..1 << fold_challenges.len()];
+
+	if log_batch_size == 0 {
+		iter::zip(&mut *scratch_buffer, P::iter_slice(values)).for_each(|(dst, val)| *dst = val);
```

Comment:
btw, i confirmed (by testing) that this case can now be removed. haven't tested the impact on performance